### PR TITLE
Implement fprintf

### PR DESCRIPTION
**Pull request by lgritz**
Has always been in the docs. I swear I had this working at one point.
But it seems to be MIA.

### CODE REVIEW

File: src/liboslexec/opstring.cpp [hunk 1]
```diff
@@ -213,6 +213,24 @@ osl_warning (ShaderGlobals *sg, const char* format_str, ...)
...
+OSL_SHADEOP void
+osl_fprintf (ShaderGlobals *sg, const char *filename,
+             const char* format_str, ...)
+{
+    va_list args;
+    va_start (args, format_str);
+    std::string s = Strutil::vformat (format_str, args);
+    va_end (args);
+
+    OIIO::mutex fprintf_mutex;
```
Comment (fpsunflower) [inline src/liboslexec/opstring.cpp]: Did you mean to declare this mutex as `static` ?

Comment (lgritz) [inline src/liboslexec/opstring.cpp]: Yes, I certainly did. :-)

Comment (lgritz) [inline src/liboslexec/opstring.cpp]: Pushed an update to fix this.

**Comment by fpsunflower** [comment]
LGTM

**Pull request merged**